src/internal/pfsdb/branches.go

```diff
@@ -60,28 +66,42 @@ type BranchInfoWithID struct {
 	*pfs.BranchInfo
 }
 
-func NewBranchIterator(ctx context.Context, tx *pachsql.Tx, startPage, pageSize uint64, project, repo, repoType string, sortOrder sortOrder) (*BranchIterator, error) {
+func NewBranchIterator(ctx context.Context, tx *pachsql.Tx, startPage, pageSize uint64, filter *pfs.Branch, orderByColumns []string, order sortOrder) (*BranchIterator, error) {
```

Comment:
I think you should use a custom type here for the columns. The type itself will be unexported but the constants of that type will be exported, that should create some safety and prevent people from ordering with arbitrary columns.